### DISCUSSION TITLE
Support Google Artifact Registry Authentication with Go?

### DISCUSSION THREAD

**Discussion by Kiharu-Ishikawa** [renovatebot/renovate#30567] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github self hosted runner

### Please tell us more about your question or problem

Instructions on how to use GAR package with Go can be found at
https://cloud.google.com/artifact-registry/docs/go/manage-modules?hl=en

According to the above there is a way to authenticate with .netrc.
I could not find any description about authentication with .netrc in renovate docs.
Is there any way to authenticate GAR when renovate get a private Go package

I tried to pass short lived token like the authentication with private npm package.
It did not work.

- Log
```
"cmd": "/bin/sh -c go get -d -t ./...",
"stderr": "go: reading https://${my_registry}/${my_module}/@v/vx.x.x.mod: 401 Unauthorized\n\tserver response: The request does not have valid authentication credentials. 
```

my config.js is like

```
hostRules: [
  {
      matchHost: 'my_registry/*',
      token: process.env.AUTH_TOKEN,
  },
],

secrets: {
    AUTH_TOKEN: process.env.RENOVATE_AUTH_TOKEN,
},
```

**Comment by adavid-panw** [comment]
@Kiharu-Ishikawa did you manage to work it out? I stumbled in the same error
